--- fetchmail-6.4.30/imap.c	2021-10-31 14:54:24.000000000 +0300
+++ fetchmail-6.4.30/imap-less-timeout.c	2022-06-28 18:00:26.429207632 +0300
@@ -793,7 +793,7 @@
 	/* special timeout to terminate the IDLE and re-issue it
 	 * at least every 28 minutes:
 	 * (the server may have an inactivity timeout) */
-	mytimeout = idle_timeout = 1680; /* 28 min */
+	mytimeout = idle_timeout = 480; /* 8 min for Gmail */
 	time(&idle_start_time);
 	stage = STAGE_IDLE;
 	/* enter IDLE mode */
